### PR TITLE
Update rng_salt value to v0.44.0

### PR DESCRIPTION
**Pull request by Alex-Preciado**
**Context:**
As usual, we need to update `rng_salt` to see any vulnerable tests being broken in new release.

**Description of the Change:**
We found 8 breaking and categorized as the following two classes:

Fixed by **actually using the seeding system**
 - [x] classical shadow `test_return_distribution`

Temporary xfailed along with deligated following-up stories; each of them needs specific re-design of tolerance, according to the statistics of each (basically first calculate the theoretical standard deviation and then apply the three sigma rule).
 - [x] `test_simulate.py`: `test_simulate_one_shot_native_mcm`
 - [x] `test_pulse_gradient.py`: `test_qnode_probs_expval_broadcasting`
 - [x] `test_spsa_gradient.py`: `test_mean_and_var` for `TestRademacherSampler`
 - [x] `test_involutory_and_noninvolutory_variance_single_param`
 - [x] `test_reflection.py`: `test_qnode_jax`
 - [x] `test_jax_jit_qnode.py`: `test_finite_shot_multiple_measurements`
 - [x] `test_hermitian_edge_cases.py`: `test_hermitian_two_wires_permuted`

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-98591]

### CODE REVIEW

File: tests/devices/qubit/test_simulate.py [hunk 1]
```diff
@@ -1616,6 +1616,7 @@ def test_tree_traversal_combine_measurements(self, measurements, expected):
 
     # Near 10% failure rate; need revise and fix soon
     # FIXME: [sc-95724]
+    @pytest.mark.skip(reason="Flaky test; needs to be fixed sc-95724")
```
Comment (JerryChen97) [inline tests/devices/qubit/test_simulate.py]:
```suggestion
    @pytest.mark.xfail(reason="Flaky test; needs to be fixed sc-95724", strict=False)
```

File: tests/templates/subroutines/test_reflection.py [hunk 2]
```diff
@@ -201,9 +201,21 @@ def test_qnode_autograd(self, shots, seed):
 
         assert np.allclose(res, self.exp_jac, atol=0.005)
 
+    # @pytest.mark.skip(reason="TEMP: Test is flaky, needs investigation")
```
Comment (JerryChen97) [inline tests/templates/subroutines/test_reflection.py]:
```suggestion
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 3]
```diff
@@ -1726,6 +1723,7 @@ def test_expval_and_variance_single_param(self, broadcast):
             assert isinstance(gradF, tuple)
             assert gradF == pytest.approx(expected, abs=finite_diff_tol)
 
+    @pytest.mark.skip(reason="Flaky test, temporarily skipping for investigation.")
```
Comment (JerryChen97) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
    @pytest.mark.xfail(reason="Flaky test, to be fixed at sc-101770", strict=False)
```

File: tests/devices/qubit/test_simulate.py [hunk 4]
```diff
@@ -1616,6 +1616,7 @@ def test_tree_traversal_combine_measurements(self, measurements, expected):
 
     # Near 10% failure rate; need revise and fix soon
     # FIXME: [sc-95724]
+    @pytest.mark.xfail(reason="Flaky test; needs to be fixed sc-95724", strict=False)
```
Comment (Alex-Preciado) [inline tests/devices/qubit/test_simulate.py]:
```suggestion
    @pytest.mark.xfail(reason="Flaky test under investigation (tracked in sc-95724)", strict=False)
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 5]
```diff
@@ -1371,7 +1371,7 @@ def test_non_involutory_variance_multi_param(self, broadcast, seed):
             assert gradF[0] == pytest.approx(expected, abs=2)
             assert qml.math.allclose(gradF[1], expected, atol=1.5)
 
-    @pytest.mark.local_salt(42)
+    @pytest.mark.xfail(reason="Flaky test, to be fixed at sc-101770", strict=False)
```
Comment (Alex-Preciado) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
    @pytest.mark.xfail(reason="Flaky test under investigation (tracked in sc-101770)", strict=False)
```

File: tests/gradients/core/test_pulse_gradient.py [hunk 6]
```diff
@@ -1579,6 +1579,7 @@ def ansatz(params):
         )
         jax.clear_caches()
 
+    @pytest.mark.xfail(reason="Temp xfail. See sc-101765 for progress.", strict=False)
```
Comment (Alex-Preciado) [inline tests/gradients/core/test_pulse_gradient.py]:
```suggestion
    @pytest.mark.xfail(reason="Test seems wrong and is under investigation (tracked in sc-101765)", strict=False)
```

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 7]
```diff
@@ -1726,6 +1723,7 @@ def test_expval_and_variance_single_param(self, broadcast):
             assert isinstance(gradF, tuple)
             assert gradF == pytest.approx(expected, abs=finite_diff_tol)
 
+    @pytest.mark.xfail(reason="Flaky test, to be fixed at sc-101770", strict=False)
```
Comment (Alex-Preciado) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py]:
```suggestion
    @pytest.mark.xfail(reason="Flaky test under investigation (tracked in sc-101770)", strict=False)
```

File: tests/templates/subroutines/test_reflection.py [hunk 8]
```diff
@@ -203,7 +203,18 @@ def test_qnode_autograd(self, shots, seed):
...
     @pytest.mark.parametrize("use_jit", [False, True])
-    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.parametrize(
+        "shots",
+        [
+            None,
+            pytest.param(
+                50000,
+                marks=pytest.mark.xfail(
+                    reason="Flaky test, to be fixed at sc-101771", strict=False
```
Comment (Alex-Preciado) [inline tests/templates/subroutines/test_reflection.py]:
```suggestion
                    reason="Flaky test under investigation (tracked in sc-101771)", strict=False
```

File: tests/test_hermitian_edge_cases.py [hunk 9]
```diff
@@ -89,6 +89,7 @@ def circuit():
         expected = ((a - d) * np.cos(theta) + 2 * re_b * np.sin(theta) * np.sin(phi) + a + d) / 2
         assert np.allclose(res, expected, atol=0.01, rtol=0)
 
+    @pytest.mark.xfail(reason="atol is not reasonable. To fix at sc-101767", strict=False)
```
Comment (Alex-Preciado) [inline tests/test_hermitian_edge_cases.py]:
```suggestion
    @pytest.mark.xfail(reason="Test under investigation: atol doesn't seem reasonable (tracked in sc-101767)", strict=False)
```

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py [hunk 10]
```diff
@@ -895,6 +895,7 @@ def circuit(x):
         expected_g = 2 * np.cos(0.5) * np.sin(0.5)
         assert qml.math.allclose(g, expected_g, atol=1 / qml.math.sqrt(shots), rtol=0.03)
 
+    @pytest.mark.xfail(reason="to be fixed at sc-101768", strict=False)
```
Comment (Alex-Preciado) [inline tests/workflow/interfaces/qnode/test_jax_jit_qnode.py]:
```suggestion
    @pytest.mark.xfail(reason="Test under investigation (tracked in sc-101768)", strict=False)
```

**Comment by JerryChen97** [comment]
> Just some minor adjustments to the xfail descriptions for clarity and consistency 😎

Thanks! All committed
